packages/code/src/Code.tsx

```diff
@@ -80,6 +92,136 @@ function getWrapperVariantStyle(variant: Variant): string {
   `;
 }
 
+function getSidebarVariantStyle(variant: Variant): string {
+  const colors = variantColors[variant];
+
+  switch (variant) {
+    case 'light':
+      return css`
+        border-color: ${colors[1]};
+        background-color: white;
+      `;
+    case 'dark':
+      return css`
+        border-color: ${colors[1]};
+        background-color: ${colors[1]};
+      `;
+  }
+}
+
+function getCopyButtonStyle(variant: Variant, copied: boolean): string {
+  if (copied) {
+    return css`
+      align-self: center;
+      color: ${uiColors.white};
+      background-color: ${uiColors.green.base};
+    `;
+  }
+
+  if (variant === Variant.Dark) {
+    return css`
+      align-self: center;
+      color: ${uiColors.gray.base};
+      background-color: ${uiColors.gray.dark3};
+    `;
+  }
+
+  return css`
+    align-self: center;
+    color: ${uiColors.gray.base};
+  `;
+}
+
+const ScrollState = {
+  None: 'none',
+  Left: 'left',
+  Right: 'right',
+  Both: 'both',
+} as const;
+
+type ScrollState = typeof ScrollState[keyof typeof ScrollState];
+
+function getScrollShadowStyle(
+  scrollState: ScrollState,
+  variant: Variant,
+): string {
+  const colors = variantColors[variant];
+  const shadowColor =
+    variant === Variant.Light ? 'rgba(93,108,116,0.3)' : 'rgba(0,0,0,0.35)';
+
+  if (scrollState === ScrollState.Both) {
+    return css`
+      box-shadow: inset 6px 0 6px -6px ${shadowColor},
+        inset -6px 0 6px -6px ${shadowColor}, inset 0 6px 6px -6px ${colors[0]},
+        inset 0 -6px 6px -6px ${colors[0]};
+    `;
+  }
+
+  if (scrollState === ScrollState.Left) {
+    return css`
+      box-shadow: inset 6px 0 6px -6px ${shadowColor};
+    `;
+  }
+
+  if (scrollState === ScrollState.Right) {
+    return css`
+      box-shadow: inset -6px 0 6px -6px ${shadowColor};
+    `;
+  }
+
+  return '';
+}
+
+interface ActionType {
+  type: string;
+}
+
+interface StateType {
+  scroll: ScrollState;
+  copied: boolean;
+  isLoaded: boolean;
+}
+
+const initialState = {
+  scroll: ScrollState.None,
+  copied: false,
+  isLoaded: false,
+};
+
+function scrollReducer(state: StateType, action: ActionType) {
```

Comment:
So...seems like your refactor cleaned up the state changes we have to support, such that we're only support one state change per dispatch.

That being the case...it may make sense to go back to using useState as our state transitions are simple again and not inter-related.

I don't want to just keep pointing you in a new direction, but hopefully you understand the reasoning behind the shifts in my thinking?

Reply:
I totally understand. Refactoring back to useState was actually a pretty quick process and I think it came out a little more clear.